plugins/net_plugin/net_plugin.cpp

```diff
@@ -3899,6 +4259,8 @@ namespace eosio {
       if( is_transactions_only_connection() ) hello.p2p_address += ":trx";
       // if we are not accepting transactions tell peer we are blocks only
       if( is_blocks_only_connection() || !my_impl->p2p_accept_transactions ) hello.p2p_address += ":blk";
+      if( is_peers_only_connection() ) hello.p2p_address += ":peer";
+      if( is_all_connection() ) hello.p2p_address += ":all";
```

Comment:
This logic does not look correct. If `!my_impl->p2p_accept_transactions` then `hello.p2p_address` already contains `:blk`.

Reply:
changed the connection type enum logic.